yt/yt/server/query_tracker/handler_base.cpp

```diff
@@ -376,6 +384,31 @@ bool TQueryHandlerBase::TryWriteQueryState(EQueryState state, EQueryState previo
         }
         WaitFor(transaction->Commit())
             .ThrowOnError();
+
+        auto now = TInstant::Now();
+        StateTimes[previousState] += now - LastStateChange_.value_or(record.StartTime);
```

Comment:
This logic may sometimes give us incorrect information. For example, when a query started, changed state, and then got assigned to a different query tracker. I am not sure if this situation is possible now, but seems like it should be.

Instead of `record.StartTime` it would be more correct to use time, when a query tracker acquired the query